components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2183,69 +2183,20 @@ public Group updateGroup(Group oldGroup, Group newGroup, Map<String, Boolean> re
                 /*set thread local property to signal the downstream SCIMUserOperationListener
                 about the provisioning route.*/
             SCIMCommonUtils.setThreadLocalIsManagedThroughSCIMEP(true);
-            //check if the user ids sent in updated group exist in the user store and the associated user name
-            //also a matching one.
-            List<Object> userIds = newGroup.getMembers();
-            List<String> userDisplayNames = newGroup.getMembersWithDisplayName();
-
-                /* compare user store domain of group and user store domain of user name , if there is a mismatch do not
-                 update the group */
-            if (userDisplayNames != null && userDisplayNames.size() > 0) {
-                for (String userDisplayName : userDisplayNames) {
-                    String userStoreDomainForUser =
-                            IdentityUtil.extractDomainFromName(userDisplayName);
-                    if (!isInternalOrApplicationGroup(userStoreDomainForGroup) && !userStoreDomainForGroup.equalsIgnoreCase
-                            (userStoreDomainForUser)) {
-                        throw new IdentitySCIMException(
-                                userDisplayName + " does not " + "belongs to user store " + userStoreDomainForGroup);
-                    }
 
-                }
+            // Find out added user's user ids as a list.
+            List<Object> newlyAddedUserIds = newGroup.getMembers();
+            List<Object> oldGroupUserIds = oldGroup.getMembers();
```

Comment:
Can't oldGroupUserIds be null? If it can be null then when you call "oldGroupUserIds.size()" NPE will be thrown

Reply:
Fixed